connection_configuration.go

```diff
@@ -0,0 +1,468 @@
+				err.MessageArgs = []interface{}{key, value}
+				return err
+			}
+		case "passcodeinpassword":
+			if cfg.PasscodeInPassword, parsingErr = parseBool(value); parsingErr != nil {
+				err.MessageArgs = []interface{}{key, value}
+				return err
+			}
+		case "clienttimeout":
+			if cfg.ClientTimeout, parsingErr = parseDuration(value); parsingErr != nil {
+				err.MessageArgs = []interface{}{key, value}
+				return err
+			}
+		case "jwtclienttimeout":
+			if cfg.JWTClientTimeout, parsingErr = parseDuration(value); parsingErr != nil {
+				err.MessageArgs = []interface{}{key, value}
+				return err
+			}
+		case "logintimeout":
+			if cfg.LoginTimeout, parsingErr = parseDuration(value); parsingErr != nil {
+				err.MessageArgs = []interface{}{key, value}
+				return err
+			}
+		case "requesttimeout":
+			if cfg.RequestTimeout, parsingErr = parseDuration(value); parsingErr != nil {
+				err.MessageArgs = []interface{}{key, value}
+				return err
+			}
+		case "jwttimeout":
+			if cfg.JWTExpireTimeout, parsingErr = parseDuration(value); parsingErr != nil {
+				err.MessageArgs = []interface{}{key, value}
+				return err
+			}
+		case "externalbrowsertimeout":
+			if cfg.ExternalBrowserTimeout, parsingErr = parseDuration(value); parsingErr != nil {
+				err.MessageArgs = []interface{}{key, value}
+				return err
+			}
+		case "maxretrycount":
+			if cfg.MaxRetryCount, parsingErr = parseInt(value); parsingErr != nil {
+				err.MessageArgs = []interface{}{key, value}
+				return err
+			}
+		case "application":
+			cfg.Application, parsingErr = parseString(value)
+			if parsingErr != nil {
+				err.MessageArgs = []interface{}{key, value}
+				return err
+			}
+		case "authenticator":
+			v, parsingErr = parseString(value)
+			if parsingErr != nil {
+				err.MessageArgs = []interface{}{key, value}
+				return err
+			}
+			parsingErr = determineAuthenticatorType(cfg, v)
+			if parsingErr != nil {
+				err.MessageArgs = []interface{}{key, value}
+				return err
+			}
+		case "insecuremode":
+			if cfg.InsecureMode, parsingErr = parseBool(value); parsingErr != nil {
+				err.MessageArgs = []interface{}{key, value}
+				return err
+			}
+		case "ocspfailopen":
+			if vv, parsingErr = parseBool(value); parsingErr != nil {
+				err.MessageArgs = []interface{}{key, value}
+				return err
+			}
+			if vv {
+				cfg.OCSPFailOpen = OCSPFailOpenTrue
+			} else {
+				cfg.OCSPFailOpen = OCSPFailOpenFalse
+			}
+
+		case "token":
+			cfg.Token, parsingErr = parseString(value)
+			if parsingErr != nil {
+				err.MessageArgs = []interface{}{key, value}
+				return err
+			}
+		case "privatekey":
+			v, parsingErr = parseString(value)
+			if parsingErr != nil {
+				err.MessageArgs = []interface{}{key, value}
+				return err
+			}
+			block, decodeErr := base64.URLEncoding.DecodeString(v)
+			if decodeErr != nil {
+				err = &SnowflakeError{
+					Number:  ErrCodePrivateKeyParseError,
+					Message: "Base64 decode failed",
+				}
+				return err
```

Comment:
Same here. Return the error directly.

Reply:
Done.

Reply:
Sorry, why can't we return decodeErr instead? Why does it need a custom error here?